### DISCUSSION TITLE
Websocket data sending example

### DISCUSSION THREAD

**Discussion by torcsvarizs-ar** [AxisCommunications/acap3-examples#203] [Q&A] [answered]
Hello, are there examples for sending data from the camera to a webpage on registered websocket channel? Are we allowed to upgrade from http request to websocket from a request handler based on ax_http api? Thanks!

**Comment by pataxis** [comment]
Hi @torcsvarizs-ar, thanks for your question, we'll get back to you when we have had time to look in to this.

**Reply by torcsvarizs-ar** [reply]
Thank you!

**Reply by pataxis** [reply, marked as answer]
> are there examples for sending data from the camera to a webpage on registered websocket channel?

None that I know of.

> Are we allowed to upgrade from http request to websocket from a request handler based on ax_http api?

As far as I understand the library, it's mostly redirecting requests over HTTP, or as the library says itself 
_The AXHttp library provides a simple mechanism for serving CGI requests._

I'm not sure if it would be technically possible to serve over WebSockets, nor if it's recommended.


In [acap-native-sdk-examples](https://github.com/AxisCommunications/acap-native-sdk-examples), there is an example called [web-server](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/web-server) that has replaced AxHTTP. I'm not sure if that can handle more of what you want to do. I think it should be possible to use also in ACAP 3, even though it's not confirmed.
There is also an example with FastCGI but from reading up on WebSockets, it seems like they are not compatible.

If you are developing a new ACAP application, it's recommended to start with acap-native-sdk-examples that get regular SDK updates instead. If you want to know more you can find links to differences [here](https://github.com/orgs/AxisCommunications/discussions/1) and the different repos [here](https://github.com/orgs/AxisCommunications/discussions/125).
